code/game/machinery/washing_machine.dm

```diff
@@ -238,102 +307,160 @@
-			to_chat(user, "This item does not fit.")
-			return
-		if( istype(W,/obj/item/clothing/mask/gas ) )
-			to_chat(user, "This item does not fit.")
-			return
-		if( istype(W,/obj/item/clothing/mask/cigarette ) )
-			to_chat(user, "This item does not fit.")
-			return
-		if( istype(W,/obj/item/clothing/head/syndicatefake ) )
-			to_chat(user, "This item does not fit.")
-			return
-//		if( istype(W,/obj/item/clothing/head/powered ) )
-//			to_chat(user, "This item does not fit.")
-//			return
-		if( istype(W,/obj/item/clothing/head/helmet ) )
-			to_chat(user, "This item does not fit.")
-			return
-		if( istype(W,/obj/item/clothing/gloves/furgloves ) )
-			to_chat(user, "This item does not fit.")
-			return
+		var/list/prohibited = list(/obj/item/clothing/suit/space, /obj/item/clothing/suit/syndicatefake, /obj/item/clothing/suit/bomb_suit, /obj/item/clothing/suit/armor, /obj/item/clothing/mask/gas, /obj/item/clothing/head/syndicatefake, /obj/item/clothing/head/helmet, /obj/item/clothing/suit/cyborg_suit, /obj/item/clothing/mask/cigarette)
+		if(!istype(W, (/obj/item/clothing/head/helmet/space/plasmaman)))
+			for(var/I in 1 to length(prohibited))
+				if(istype(W, prohibited[I]))
+					to_chat(user, "This item does not fit.")
+					return
 		if(istype(W, /obj/item/clothing/gloves/color/black/krav_maga/sec))
 			to_chat(user, "<span class='warning'>Washing these gloves would fry the electronics!</span>")
 			return
 		if(W.flags & NODROP) //if "can't drop" item
 			to_chat(user, "<span class='notice'>\The [W] is stuck to your hand, you cannot put it in the washing machine!</span>")
 			return
 
-		if(contents.len < 5)
-			if( state in list(1, 3) )
-				user.drop_item()
-				W.loc = src
-				state = 3
+		if(contents.len < 10)
+			if(!locate(/mob,contents)) // no placing items in the machine if theres a mob inside
+				if(open && !bloody)
+					user.drop_item()
+					W.loc = src
+					full = TRUE
+				else
+					to_chat(user, "<span class='notice'>You can't put the item in right now.</span>")
 			else
-				to_chat(user, "<span class='notice'>You can't put the item in right now.</span>")
+				to_chat(user, "<span class='notice'>The washing machine is is occupied by [locate(/mob,contents)], there's no extra space.</span>")
 		else
 			to_chat(user, "<span class='notice'>The washing machine is full.</span>")
 		update_icon()
 	else
 		return ..()
 
 /obj/machinery/washing_machine/attack_hand(mob/user as mob)
-	switch(state)
-		if(1)
-			state = 2
-		if(2)
-			state = 1
-			for(var/atom/movable/O in contents)
-				O.loc = src.loc
-		if(3)
-			state = 4
-		if(4)
-			state = 3
-			for(var/atom/movable/O in contents)
-				O.loc = src.loc
-			crayon = null
-			state = 1
-		if(5)
-			to_chat(user, "<span class='warning'>[src] is busy.</span>")
-		if(6)
-			state = 7
-		if(7)
-			if(gibs_ready)
-				gibs_ready = 0
+	if(running) // States 5 and 8
+		to_chat(usr, "<span class='warning'>\The [src] is busy.</span>")
+	else
```

Comment:
Would be really nice to have an early return in the if and unindent the whole block underneath